### DISCUSSION TITLE
Forgot admin credentials

### DISCUSSION THREAD

**Discussion by DAVIDOJONUGWAMOSES** [keycloak/keycloak#14063] [Q&A]
### Description

I created admin but when I tried to login, it is telling me invalid username or password. Can I create another admin?

**Comment by fredericoschardong** [comment]
Sometimes this happens to me, and I have no idea why or how to reproduce.

**Reply by kkcmadhu** [reply]
you can use the regular, forgot password stuff for resetting your admin password too!

if you are in k8s etc, you can refer your secret or env variable where you would have seeded the admin user name password.

**Reply by fredericoschardong** [reply]
I was able to narrow it down to SMTP settings. On keycloak 19.0.3, if I enter invalid SMTP settings and leave the admin's e-mail address blank, I can no longer login as admin.

**Reply by kkcmadhu** [reply]
you can simply change the usnername and password in your env variable, it will create a new admin.
refer docs here https://www.keycloak.org/server/configuration#_setup_of_the_initial_admin_user

something like this

   env:
        - name: KEYCLOAK_ADMIN
          value: "youradminusername"
        - name: KEYCLOAK_ADMIN_PASSWORD
          value: "YourAdminUsersPassoword!!"

**Reply by sriharsha-bhasuru** [reply]
@fredericoschardong  happened to me today.

**Reply by 23b** [reply]
I also messed with the SMTP settings. In this process you will be asked to set a email for your admin user. By mistake i also changed the admin username to its email address. Only found out after [connecting to the H2 database](https://www.mastertheboss.com/keycloak/how-to-access-keycloak-h2-database/) to delete the admin user for a password reset. Turned out I just had to enter the email as username...

**Comment by DAVIDOJONUGWAMOSES** [comment]
Thank you very much
